src/persist/src/indexed/encoding.rs

```diff
@@ -92,6 +92,14 @@ pub struct BlobMeta {
     ///
     /// Invariant: Each stream id is in here at most once.
     pub traces: Vec<TraceMeta>,
+    /// Unsealed blobs to delete.
+    ///
+    /// Invariant: Each key is in here or pending_trace_deletes at most once.
+    pub pending_unsealed_deletes: Vec<UnsealedBatchMeta>,
```

Comment:
I'm on the fence about whether we actually should do anything about this - but we only really need the key and the size of the batch to delete. its a shame that we duplicate the full batch metadata here and for traces